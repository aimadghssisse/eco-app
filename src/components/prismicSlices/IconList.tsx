/**
 * Prismic Slice - Landing page Badge List
 */

import * as React from "react"

const IconList = ({ data }) => {
    return (
        <div className="flex flex-wrap justify-center mt-12 px-6">
            {data.items.map((icon, index) => (
                <div key={index} className="flex-1 flex-auto flex-col items-center justify-center text-center mb-16">
                    <img className="h-10 mb-4 inline-block" src={icon.icon.url} alt={icon.icon.alt}/>
                    <p>{icon.text}</p>
                </div>
            ))}
        </div>
    )
}

export default IconList