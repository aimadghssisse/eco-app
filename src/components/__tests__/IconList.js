import React from "react"
import renderer from "react-test-renderer"

import IconList from "../prismicSlices/IconList"

describe("IconList", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(
                <div className="flex flex-wrap justify-center my-12">
                    <div key="key" className="flex-1 flex flex-col items-center justify-center mx-1 mb-9 min-w-max">
                        <img className="icon-height42 mb-4" src="icon.icon.url" alt="icon.icon.alt" />
                        <p>icon.text</p>
                    </div>
                </div>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})