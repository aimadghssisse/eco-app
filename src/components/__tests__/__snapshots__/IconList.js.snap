// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconList renders correctly 1`] = `
<div
  className="flex flex-wrap justify-center my-12"
>
  <div
    className="flex-1 flex flex-col items-center justify-center mx-1 mb-9 min-w-max"
  >
    <img
      alt="icon.icon.alt"
      className="icon-height42 mb-4"
      src="icon.icon.url"
    />
    <p>
      icon.text
    </p>
  </div>
</div>
`;

exports[`IconList renders correctly 1`] = `
<div
  className="flex flex-wrap justify-center my-12"
>
  <div
    className="flex-1 flex flex-col items-center justify-center mx-1 mb-9 min-w-max"
  >
    <img
      alt="icon.icon.alt"
      className="icon-height42 mb-4"
      src="icon.icon.url"
    />
    <p>
      icon.text
    </p>
  </div>
</div>
`;
